appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroViewPager.kt

```diff
@@ -111,6 +117,25 @@ class AppIntroViewPager(context: Context, attrs: AttributeSet) : ViewPager(conte
             MotionEvent.ACTION_UP -> performClick()
             MotionEvent.ACTION_DOWN -> currentTouchDownX = event.x
             MotionEvent.ACTION_MOVE -> {
+
+                // If the slide contains permissions, check for forward swipe.
+                if (isPermissionSlide) {
+                    if (isSwipeForward(currentTouchDownX, event.x))
+                    // If the swipe is occured before 200ms, dont do anything.
+                    {
+                        if (System.currentTimeMillis() - permDialogSwipeLastCalled >= ON_PERMISSION_DIALOG_MAX_INTERVAL) {
+                            onNextPageRequestedListener?.onUserRequestedPermissionsDialog()
+                            LogHelper.d("Pager", "Permisson being requested  ")
+                            return isPagingEnabled
+                        } else {
+                            return false
+                        }
+                    }
+                } else {
+                    isPagingEnabled = true
+                }
+
+                permDialogSwipeLastCalled = System.currentTimeMillis()
```

Comment:
No, it's not, (BTW Which line are you talking about?)

Reply:
> it doesn't work for fast swipe on the first slide if placed inside the conditional

That means that we have another bug :/

Reply:
Not exactly, this is bound to happen given that we cannot stop input to a viewpager while it is swiping.
E.g. we have a permission on the second slide, we swipe on the first slide and then quickly swipe again. Since the first slide didn't contain permissions, if this line is placed inside the conditional, permDialogswipelastcalled doesn't get a value, and after the second quick swipe, the permission dialog opens, but if we let the code remain as it is, we don't have this bug.

Reply:
Lemme take a look still, we can still do some refactoring